crates/qt-build-utils/src/lib.rs

```diff
@@ -448,12 +480,21 @@ impl QtBuild {
     /// Get the include paths for Qt, including Qt module subdirectories. This is intended
     /// to be passed to whichever tool you are using to invoke the C++ compiler.
     pub fn include_paths(&self) -> Vec<PathBuf> {
-        let root_path = self.qmake_query("QT_INSTALL_HEADERS");
         let mut paths = Vec::new();
+
+        if self.has_framework_libs {
+            let lib_root_path: String = self.qmake_query("QT_INSTALL_LIBS");
+
+            for qt_module in &self.qt_modules {
+                paths.push(format!("{lib_root_path}/Qt{qt_module}.framework/Headers"));
+            }
+        }
+        let root_path: String = self.qmake_query("QT_INSTALL_HEADERS");
         for qt_module in &self.qt_modules {
             paths.push(format!("{root_path}/Qt{qt_module}"));
         }
```

Comment:
Is this correct when Qt is installed as a framework? This allows using include paths like `#include <QString>` instead of `#include <QtCore/QString>`. I would think we'd need to add `{lib_root_path}/Qt{qt_module}.framework/Headers/Qt{qt_module}` in the framework case.

Reply:
I never included using `#include <QtCore/QString>`, always using `#include <QString>` like it's shown in the Qt documentation. So, I don't see the problem. Or I misunderstand your comment ?

Reply:
I think this code should likely be moved into an `else` block of the `if self.has_framework_libs` statement. If it wasn't sufficient before to get the include paths with frameworks, it's probably okay to not add these include paths when using frameworks.

Reply:
I did a few researches :
The module-style includes are essential in MacOS framework style. Using the old #include <QtCore/qobject.h> or #include <QtCore/QObject> is not possible any longer. The old style still runs well on Windows and Linux thought.

I was testing only on qml_minimal example project, so this problem didn't appear.
On a hunch, I ran `QMAKE=/Users/cyril/Qt/6.5.2/macos/bin/qmake cargo run -p qml-minimal-no-cmake` , so Qt's official install on MacOs with framework style

It gave me these errors:
  cargo:warning=In file included from /Users/cyril/Devel/cxx-qt/target/debug/build/cxx-qt-lib-c19809693f328b69/out/cxxbridge/sources/cxx-qt-lib/src/core/qlist/qlist_i32.rs.cc:1:
  cargo:warning=/Users/cyril/Devel/cxx-qt/target/debug/build/cxx-qt-lib-c19809693f328b69/out/cxx-qt-lib/qlist.h:11:10: fatal error: 'QtCore/QList' file not found
  cargo:warning=#include <QtCore/QList>
  cargo:warning=         ^~~~~~~~~~~~~~
  cargo:warning=1 error generated.
  exit status: 1

Reply:
Huh curious, i found doing `#include <QtCore/QObject>` useful as it allows you to quickly determine which modules you need to include and link or why a library needs eg QtQuick as you can search for it quickly.

Guess we'll have to drop doing that if this doesn't work in macOS anymore (i can create a separate PR to do this), unless there is a different include folder that can be used that does include the module name + the header name ?

Reply:
Don't be hasty, the truth can be more complex that my explanation :D I just fixed all the QtCore, QtGui, QtQml in your lib. Guess what ? Qt's itself is floundering :
```
  exit status: 1
  cargo:warning=In file included from /Users/cyril/Devel/cxx-qt/target/debug/build/cxx-qt-lib-c19809693f328b69/out/cxxbridge/sources/cxx-qt-lib/src/core/qlist/qlist_i32.rs.cc:1:
  cargo:warning=In file included from /Users/cyril/Devel/cxx-qt/target/debug/build/cxx-qt-lib-c19809693f328b69/out/cxx-qt-lib/qlist.h:11:
  cargo:warning=In file included from /Users/cyril/Qt/6.5.2/macos/lib/QtCore.framework/Headers/QList:1:
  cargo:warning=/Users/cyril/Qt/6.5.2/macos/lib/QtCore.framework/Headers/qlist.h:8:10: fatal error: 'QtCore/qarraydatapointer.h' file not found
  cargo:warning=#include <QtCore/qarraydatapointer.h>
  cargo:warning=         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=1 error generated.
  exit status: 1
```

So, something is lacking.

Reply:
OK, i'll hold off that mega change for now :sweat_smile: And as it looks like Qt itself is doing the same, looks like it should be supported which is good :-D

Reply:
I'm wondering if there isn't a link with the "-framework [Qt module]" argument. My clang doesn't have it, yet it's a common argument for frameworks on the internet.

Reply:
> I'm wondering if there isn't a link with the "-framework [Qt module]" argument. My clang doesn't have it, yet it's a common argument for frameworks on the internet.

`parse_libs_cflags` should already take care of linking as a framework if the `.prl` files have those `-framework` or `-F` linker arguments. That's separate from include paths though.

Reply:
> I did a few researches

Okay, this is helpful information. Let's move
```rust
        let root_path: String = self.qmake_query("QT_INSTALL_HEADERS");
        for qt_module in &self.qt_modules {
            paths.push(format!("{root_path}/Qt{qt_module}"));
        }
        paths.push(root_path);
```
into an `else` block of the `if self.has_framework_libs` statement because that code doesn't work for frameworks.

Regardless, we still have the question of how to support include paths like `#include <QtCore/QString>` with frameworks. Can you search around the Qt installations for a directory called `QtCore`? If we can add the parent directory of that as an include path, hopefully all the includes will work. Also, can you post the output of `qmake -query` with the framework installation of Qt?

Reply:
Sorry, I'm holidaying. 

```
cyril@MacBook-Pro-de-Cyril cxx-qt % /Users/cyril/Qt/6.5.2/macos/bin/qmake -v
QMake version 3.1
Using Qt version 6.5.2 in /Users/cyril/Qt/6.5.2/macos/lib
cyril@MacBook-Pro-de-Cyril cxx-qt % /Users/cyril/Qt/6.5.2/macos/bin/qmake -query
QT_SYSROOT:
QT_INSTALL_PREFIX:/Users/cyril/Qt/6.5.2/macos
QT_INSTALL_ARCHDATA:/Users/cyril/Qt/6.5.2/macos
QT_INSTALL_DATA:/Users/cyril/Qt/6.5.2/macos
QT_INSTALL_DOCS:/Users/cyril/Qt/Docs/Qt-6.5.2
QT_INSTALL_HEADERS:/Users/cyril/Qt/6.5.2/macos/include
QT_INSTALL_LIBS:/Users/cyril/Qt/6.5.2/macos/lib
QT_INSTALL_LIBEXECS:/Users/cyril/Qt/6.5.2/macos/libexec
QT_INSTALL_BINS:/Users/cyril/Qt/6.5.2/macos/bin
QT_INSTALL_TESTS:/Users/cyril/Qt/6.5.2/macos/tests
QT_INSTALL_PLUGINS:/Users/cyril/Qt/6.5.2/macos/plugins
QT_INSTALL_QML:/Users/cyril/Qt/6.5.2/macos/qml
QT_INSTALL_TRANSLATIONS:/Users/cyril/Qt/6.5.2/macos/translations
QT_INSTALL_CONFIGURATION:
QT_INSTALL_EXAMPLES:/Users/cyril/Qt/Examples/Qt-6.5.2
QT_INSTALL_DEMOS:/Users/cyril/Qt/Examples/Qt-6.5.2
QT_HOST_PREFIX:/Users/cyril/Qt/6.5.2/macos
QT_HOST_DATA:/Users/cyril/Qt/6.5.2/macos
QT_HOST_BINS:/Users/cyril/Qt/6.5.2/macos/bin
QT_HOST_LIBEXECS:/Users/cyril/Qt/6.5.2/macos/libexec
QT_HOST_LIBS:/Users/cyril/Qt/6.5.2/macos/lib
QMAKE_SPEC:macx-clang
QMAKE_XSPEC:macx-clang
QMAKE_VERSION:3.1
QT_VERSION:6.5.2
```

Reply:
[qt_files.txt](https://github.com/KDAB/cxx-qt/files/12284284/qt_files.txt)

Reply:
I looked at the CI output from another projecting using the Qt official binary installer and the include paths are:
```
-isystem /Users/build/Qt/online/6.3.2/macos/lib/QtCore.framework/Headers
-isystem /Users/build/Qt/online/6.3.2/macos/lib/QtCore.framework/Headers/6.3.2
-isystem /Users/build/Qt/online/6.3.2/macos/lib/QtCore.framework/Headers/6.3.2/QtCore
```

So it seems we need:
```
{lib_root_path}/Qt{qt_module}.framework/Headers
{lib_root_path}/Qt{qt_module}.framework/Headers/{version}
{lib_root_path}/Qt{qt_module}.framework/Headers/{version}/{qt_module}
```

Reply:
Hello,
I'm back, I'll be able to test that this week.

Reply:
@jacquetc Have you had a chance to test the code I [posted above](https://github.com/KDAB/cxx-qt/pull/630#discussion_r1286329630)?

Reply:
No yet, I'll take time tomorrow at noon

Reply:
Tested, exactly the same error.